#### data taken from
https://www.datafiles.samhsa.gov/study-dataset/national-survey-drug-use-and-health-2016-nsduh-2016-ds0001-nid17185

In [0]:
import pandas as pd
from sklearn.model_selection import train_test_split

In [0]:
data = pd.read_csv('NSDUH_2016_Tab.tsv', sep='\t')

/home/maryana/anaconda3/lib/python3.6/site-packages/IPython/core/interactiveshell.py:2785: DtypeWarning: Columns (2506) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


In [0]:
data.describe()

QUESTID2       CIGEVER      CIGOFRSM      CIGWILYR        CIGTRY  \
count  5.689700e+04  56897.000000  56897.000000  56897.000000  56897.000000   
mean   5.433849e+07      1.512751     78.094240     78.102571    522.148883   
std    2.548617e+07      0.499842     39.396524     39.382054    487.203296   
min    1.000061e+07      1.000000      1.000000      1.000000      1.000000   
25%    3.230581e+07      1.000000     99.000000     99.000000     16.000000   
50%    5.399291e+07      2.000000     99.000000     99.000000    991.000000   
75%    7.585681e+07      2.000000     99.000000     99.000000    991.000000   
max    9.999840e+07      2.000000     99.000000     99.000000    997.000000   

             CIGYFU        CIGMFU        CIGREC      CIG30USE       CG30EST  \
count  56897.000000  56897.000000  56897.000000  56897.000000  56897.000000   
mean    9828.541347     93.001828     47.982512     78.531399     92.982003   
std     1139.541074     12.876577     44.156738     28.099777      4.322806   
min     2014.000000      1.000000      1.000000      1.000000      1.000000   
25%     9991.000000     91.000000      3.000000     91.000000     91.000000   
50%     9991.000000     91.000000     91.000000     91.000000     91.000000   
75%     9999.000000     99.000000     91.000000     93.000000     93.000000   
max     9999.000000     99.000000     91.000000     98.000000     99.000000   

           ...           POVERTY3       TOOLONG      TROUBUND        PDEN10  \
count      ...       56438.000000  56897.000000  56897.000000  56897.000000   
mean       ...           2.387115      2.152170      2.168867      1.654112   
std        ...           0.794384      4.557856      4.555436      0.618668   
min        ...           1.000000      1.000000      1.000000      1.000000   
25%        ...           2.000000      2.000000      2.000000      1.000000   
50%        ...           3.000000      2.000000      2.000000      2.000000   
75%        ...           3.000000      2.000000      2.000000      2.000000   
max        ...           3.000000     98.000000     98.000000      3.000000   

            COUTYP4      MAIIN102      AIIND102      ANALWT_C         VESTR  \
count  56897.000000  56897.000000  56897.000000  56897.000000  56897.000000   
mean       1.747561      1.980632      1.980737   4735.401431  40025.633654   
std        0.764244      0.137817      0.137449   5635.587553     14.518543   
min        1.000000      1.000000      1.000000      3.846911  40001.000000   
25%        1.000000      2.000000      2.000000   1279.961222  40013.000000   
50%        2.000000      2.000000      2.000000   2853.580363  40026.000000   
75%        2.000000      2.000000      2.000000   5833.049708  40038.000000   
max        3.000000      2.000000      2.000000  84358.633589  40050.000000   

              VEREP  
count  56897.000000  
mean       1.493312  
std        0.499960  
min        1.000000  
25%        1.000000  
50%        1.000000  
75%        2.000000  
max        2.000000  

[8 rows x 2666 columns]

In [0]:
data.head(5)

QUESTID2    FILEDATE  CIGEVER  CIGOFRSM  CIGWILYR  CIGTRY  CIGYFU  CIGMFU  \
0  11635143  02/28/2018        1        99        99      16    9999      99   
1  36845143  02/28/2018        1        99        99      15    9999      99   
2  35755143  02/28/2018        1        99        99      26    9999      99   
3  94475143  02/28/2018        2         4         4     991    9991      91   
4  92675143  02/28/2018        1        99        99       5    9999      99   

   CIGREC  CIG30USE  ...    POVERTY3  TOOLONG  TROUBUND  PDEN10  COUTYP4  \
0       4        93  ...         3.0        2         2       3        3   
1       1         7  ...         3.0        1         2       2        2   
2       1         7  ...         2.0        2         2       1        1   
3      91        91  ...         3.0        2         2       1        1   
4       4        93  ...         3.0        2         2       1        1   

   MAIIN102  AIIND102      ANALWT_C  VESTR  VEREP  
0         2         2    819.434247  40037      2  
1         2         2    280.624352  40013      2  
2         2         2  10133.833583  40036      2  
3         2         2   2284.717175  40028      1  
4         2         2  24815.892373  40009      2  

[5 rows x 2668 columns]

In [0]:
rest_data, small_data = train_test_split(data, test_size = 0.05)

In [0]:
import pandas_profiling as pp
pp.ProfileReport(small_data)

In [0]:
data.shape

(56897, 2668)

91 or 991 or 9991, etc. = NEVER USED [DRUG(s) OF INTEREST]

93 or 993 or 9993, etc. = USED [DRUG] BUT NOT IN THE PERIOD OF INTEREST

94 or 994 or 9994, etc. = DON'T KNOW

97 or 997 or 9997, etc. = REFUSED

98 or 998 or 9998, etc. = BLANK (i.e., not answered; not asked the question)

99 or 999 or 9999, etc. = LEGITIMATE SKIP

94-7-8 - missing

81 (or 981, 9981, etc.) = NEVER USED [DRUG(s) OF INTEREST] Logicall yassigned

83 (or 983, 9983, etc.) = USED [DRUG] BUT NOT IN THE PERIOD OF INTEREST Logically assigned

85 (or 985, 9985, etc.) = BAD DATA Logically assigned (i.e., usually inconsistent with other data)

89 (or 989, or 9989, etc.) = LEGITIMATE SKIP Logically assigned

'IR' 'IMP REV' 'IMPUTATION REVISITED' - indicators for missing values except for IRSEX

"IMPUTATION INDICATOR" 'IMP IND' 'II' "II2"- Associated indicator variables tell which values are imputed and which ones are not

KeyError: 'STATE'